client/src/hooks/useSetAllMemberList.tsx

```diff
@@ -0,0 +1,127 @@
+
+import {useFetch} from '@apis/useFetch';
+
+import useEventId from './useEventId/useEventId';
+import {useStepList} from './useStepList/useStepList';
+
+interface UseSetAllMemberListProps {
+  validateFunc: (name: string) => ValidateResult;
+  allMemberList: string[];
+  handleCloseAllMemberListModal: () => void;
+}
+
+const useSetAllMemberList = ({
+  validateFunc,
+  allMemberList,
+  handleCloseAllMemberListModal,
+}: UseSetAllMemberListProps) => {
+  const [editedAllMemberList, setEditedAllMemberList] = useState<string[]>(allMemberList);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [errorIndexList, setErrorIndexList] = useState<number[]>([]);
+  const [canSubmit, setCanSubmit] = useState(false);
+
+  const {refreshStepList} = useStepList();
+  const {eventId} = useEventId();
+  const {fetch} = useFetch();
+
+  useEffect(() => {
+    if (arraysEqual(editedAllMemberList, allMemberList)) {
+      setCanSubmit(false);
+    } else {
+      setCanSubmit(true);
+    }
+  }, [editedAllMemberList]);
+
+  const arraysEqual = (arr1: string[], arr2: string[]) => {
+    if (arr1.length !== arr2.length) return false;
+    for (let i = 0; i < arr1.length; i++) {
+      if (arr1[i] !== arr2[i]) return false;
+    }
+    return true;
+  };
+
+  const handleNameChange = (index: number, event: React.ChangeEvent<HTMLInputElement>) => {
+    const {value} = event.target;
+    const {isValid, errorMessage: validationResultMessage} = validateFunc(value);
+
+    if (isValid && value.length !== 0) {
+      setErrorMessage('');
+
+      setEditedAllMemberList(prev => {
+        const newList = [...prev];
+        newList[index] = value;
+        return newList;
+      });
+
+      setErrorIndexList(prev => prev.filter(i => i !== index));
+
+      setCanSubmit(true);
+    } else if (value.length === 0) {
+      setErrorMessage('');
+
+      setEditedAllMemberList(prev => {
+        const newList = [...prev];
+        newList[index] = value;
+        return newList;
+      });
+
+      changeErrorIndex(index);
+    } else {
+      setErrorMessage(validationResultMessage ?? '');
+
+      changeErrorIndex(index);
+    }
+  };
+
+  const handleClickDeleteButton = async (index: number) => {
+    const memberToDelete = editedAllMemberList[index];
+
+    await fetch(() => requestDeleteAllMemberList({eventId, memberName: memberToDelete}));
+
+    setEditedAllMemberList(prev => [...prev.slice(0, index), ...prev.slice(index + 1)]);
+
+    refreshStepList();
+  };
+
+  const handlePutAllMemberList = async () => {
+    const editedMemberName: MemberChange[] = allMemberList
+      .map((originalName, index) => {
+        if (editedAllMemberList[index] !== originalName) {
+          return {before: originalName, after: editedAllMemberList[index]};
+        }
+        return null; // 조건에 맞지 않으면 null을 반환
+      })
+      .filter(item => item !== null); // null인 항목을 필터링하여 제거
```

Comment:
.map().filter()를 사용하는 것 보단, reduce를 사용하면 좀 더 가독성을 챙길 수 있지 않을까 싶어요~

```suggestion
const handlePutAllMemberList = async () => {
    const editedMemberName: MemberChange[] = allMemberList.reduce((acc, originalName, index) => {
        const editedName = editedAllMemberList[index];
        if (editedName !== originalName) {
            acc.push({ before: originalName, after: editedName });
        }
        return acc;
    }, []);
};```